x-pack/metricbeat/module/mssql/performance/performance.go

```diff
@@ -0,0 +1,76 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package performance
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/x-pack/metricbeat/module/mssql"
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet("mssql", "performance", New,
+		mb.DefaultMetricSet(),
+		mb.WithHostParser(mssql.HostParser))
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	mb.BaseMetricSet
+	log     *logp.Logger
+	fetcher *mssql.Fetcher
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The mssql performance metricset is beta.")
+
+	logger := logp.NewLogger("mssql.performance").With("host", base.HostData().SanitizedURI)
+
+	fetcher, err := mssql.NewFetcher(base.HostData().URI,
+		[]string{
+			`SELECT [cntr_value] as page_life_expectancy FROM sys.dm_os_performance_counters WHERE [object_name] = 'SQLServer:Buffer Manager' AND [counter_name] = 'Page life expectancy'`,
```

Comment:
I'm somehow surprised that MSSQL does not have a simpler way to fetch all performance metrics.

Reply:
Not sure I can follow. So we could run 1 query and get as a result a few hundred lines and pick out of this result, the values we like?

Reply:
Yes, the implementation is verbose and with some code duplication, but explicit.

Reply:
Now, one single query that returns most of the rows (limited by a `WHERE` clause) is run.